maven-plugin/src/main/java/org/wildfly/galleon/maven/FeatureSpecGeneratorInvoker.java

```diff
@@ -341,6 +341,9 @@ private void copyArtifact(CopyArtifact task, Map<String, Artifact> artifacts) th
         if(coords.getVersion() == null) {
             artifact = artifacts.get(artifactCoords);
         }
+        if (artifact == null) {
+            artifact = artifacts.get(coords.getGroupId() + ":" + coords.getArtifactId());
+        }
```

Comment:
Where is this necessary? Shouldn't the task config be adjusted instead?

Reply:
The Artifact in the task is: org.apache.activemq:libartemis-native-32:::so
I realize that another case could be: G:A::C:T with G:A::C=G:A:V:C:T being present in artifact-versions.properties,so we should, when C exists, lookup for G:A:C,  and only look-up for G:A when there is no C

Reply:
I have an updated fix on this so that the extension is also used for resolution: you may want to look at the last commit of https://github.com/ehsavoie/galleon-plugins/tree/jfdenise